src/camera.rs

```diff
@@ -1,43 +1,89 @@
 //! Contains useful [`Camera`](struct.Camera.html) struct and `Projections`.
 use std::ops;
 
-use cgmath::{ortho as cgmath_ortho, perspective as cgmath_perspective, Deg};
+use cgmath;
 use mint;
 
 use NodePointer;
 use object::Object;
 
+/// A camera's projection.
+pub enum Projection {
+    /// An orthographic projection.
+    Orthographic(Orthographic),
+
+    /// A perspective projection.
+    Perspective(Perspective),
+}
+
 /// Camera is used to render Scene with specific `Projection`.
-pub struct Camera<P> {
+pub struct Camera {
     pub(crate) object: Object,
+
     /// Projection parameters of this camera.
-    pub projection: P,
+    pub projection: Projection,
 }
 
-impl<P> AsRef<NodePointer> for Camera<P> {
+impl Camera {
+    /// Computes the projection matrix representing the camera's projection.
+    pub fn project(&self, aspect_ratio: f32) -> mint::ColumnMatrix4<f32> {
+        self.projection.project(aspect_ratio)
+    }
+}
+
+impl Projection {
+    /// Constructs an orthographic projection.
+    pub fn orthographic<P: Into<mint::Point2<f32>>>(
+        center: P,
+        extent_y: f32,
+        near: f32,
+        far: f32,
+    ) -> Self {
+        Projection::Orthographic(Orthographic {
+            center: center.into(),
+            extent_y,
+            near,
+            far,
+        })
+    }
+
+    /// Constructs a perspective projection.
+    pub fn perspective(
+        fov_y: f32,
+        near: f32,
```

Comment:
nit: we could use `Range<f32>`

Reply:
Actually, I've missed something here. glTF allows for an infinite projection with a near value but no far value. How should we expose this?

Possibilities
1. Use `Option`
   * `fn perspective(fov_y: f32, near: f32, far: Option<f32>)`
   * `fn orthographic(..., range: Range<f32>)` - inconsistent
2. Separate functions for each case
    * `fn finite_perspective(fov_y: f32, range: Range<f32>)`
    * `fn infinite_perspective(fov_y: f32, range: RangeFrom<f32>)`
    * `fn orthographic(..., range: Range<f32>)`
3. Make generic
    * `fn perspective<R: Into<PerspectiveRange>>(fov_y: f32, range: R)`
    * `fn orthographic(..., range: Range<f32>)`

Reply:
I went with (3). Let me know what you think. :smile: